syfertext/doc.py

```diff
@@ -49,13 +49,44 @@ def set_attribute(self, name: str, value: object):
             value (object): value of the custom named attribute.
         """
 
-        # make sure there is no space in name as well prevent empty name
+        # make sure that the name is not empty and does not contains any spaces
         assert (
-            isinstance(name, str) and len(name) > 0 and (not (" " in name))
+            isinstance(name, str) and len(name) > 0 and (" " not in name)
         ), "Argument name should be a non-empty str type containing no spaces"
 
         setattr(self._, name, value)
 
+    def has_attribute(self, name: str):
+        """Checks if the Underscore object `self._` has an attribute `name`
+
+        Args:
+            name (str): name of the custom attribute.
+        
+        Returns:
+            attr_exists (bool): `True` if `self._.name` exists, otherwise `False`  
+        """
+
+        # make sure that the name is not empty and does not contains any spaces
```

Comment:
This test  is not required in has attribute, user can enter any string hasattr will return false even if it's empty or has space in it.